modules/gdscript/gdscript.cpp

```diff
@@ -482,6 +482,25 @@ void GDScript::_clear_doc() {
 	docs.clear();
 	doc = DocData::ClassDoc();
 }
+
+String GDScript::get_class_icon_path() const {
+	GDScriptParser parser;
+	Error err = parser.parse(source, path, false);
```

Comment:
The parent method is already cached.

Reply:
`EditorData::get_script_icon` caches whatever it can resolve, so until the cache is invalidated. Though at this moment we invalidate it very aggressively, but it's a separate problem. Of course, if this can be cached on GDScript side too that'd be great, but I have no idea how to approach that.

Reply:
> Of course, if this can be cached on GDScript side too that'd be great, but I have no idea how to approach that.

Done: https://github.com/dalexeev/godot/commit/096560f31889a0048bbdb83fa834fabfb8144124.

Reply:
@dalexeev Included your patch and added you as a co-author!